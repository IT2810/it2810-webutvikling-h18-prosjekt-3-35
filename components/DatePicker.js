import React, {
    Component
} from 'react';
import {
    Platform,
    AsyncStorage,
    StyleSheet,
    View,
    Text,
    Button,
    FlatList,
    TouchableWithoutFeedback,
} from 'react-native';
import DateTimePicker from 'react-native-modal-datetime-picker';

export class DatePicker extends Component {
    constructor(props) {
        super(props);
        let today = new Date();

        this.state = {
            isDateTimePickerVisible: false,
            date: today,
            dateList: null,
            text: "No date selected",
        }
    }

    componentDidMount = () => this.retrieveData();

    componentDidUpdate = () => this.storeData();

    storeData = async () => {
        try {
            //await AsyncStorage.clear();
            await AsyncStorage.setItem('dateList', JSON.stringify(this.state.dateList))
        } catch (error) {

        }
    }

    retrieveData = async () => {
        try {
            //await AsyncStorage.clear();
            const value = await AsyncStorage.getItem('dateList');
            this.updateList(JSON.parse(value));
        } catch (error) {}
    }

    updateList = (dateList) => {
        this.setState({
            dateList: dateList,
        });
    }

    updateChosenDate = (date) => {
        date  = new Date(date);
        this.setState({
            date: date,
            text: "" + date,
        });
    }

    showDateTimePicker = () => this.setState({
        isDateTimePickerVisible: true
    });

    hideDateTimePicker = () => this.setState({
        isDateTimePickerVisible: false
    });

    handleDatePicked = (date) => {
        console.log('A date has been picked: ', date);
        const dateList = this.state.dateList === null ? [] : this.state.dateList;
        console.log(dateList, date);
        dateList.push(date);
        this.setState({
            date: date,
            dateList: dateList,
        });
        this.updateList(dateList);
        this.hideDateTimePicker();
    };

    getDateList = () =>  {
        const data = this.state.dateList;
        const flatListData = [];
        for (date in data) {
            flatListData.push({
                key: date,
                value: "" + data[date],
            });
        }
        return (
            <FlatList
                data={flatListData}
                renderItem={({item}) =>
                    <TouchableWithoutFeedback onPress={ () => this.updateChosenDate(item.value)}>
                        <View>
                            <Text style={styles.item}>{item.value}</Text>
                        </View>
                    </TouchableWithoutFeedback>
            }/>
        )
    }

    render() {
        const flatList = this.getDateList();
        return ( <
            View style = {styles.container}>
                {flatList}

                <Text style = {styles.text}>
                    {this.state.text}
                </Text>

                <Button
                    title = {"Open Calendar"}
                    onPress = {() => this.showDateTimePicker()}
                    />

                <DateTimePicker
                    date = {this.state.date}
                    isVisible = {this.state.isDateTimePickerVisible}
                    onConfirm = {this.handleDatePicked}
                    onCancel = {this.hideDateTimePicker}
                    />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
    },
    text: {
        margin: 5,
        padding: 20,
        fontSize: 18,
        textAlign: 'center',
        backgroundColor: 'steelblue',
        borderColor: 'black',
        borderRadius: 2,
        borderColor: 'black',
        borderWidth: 1,
    },
    listView: {

    },
    item: {
        textAlign: 'center',
        marginTop: 4,
        padding: 10,
        borderRadius: 1,
        borderWidth: 1,
        borderColor: 'black',
        backgroundColor: 'white'
    }
});
