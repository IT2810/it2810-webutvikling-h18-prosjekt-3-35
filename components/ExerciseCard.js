import React, {
    Component
} from 'react';
import {
    StyleSheet,
    View,
    Text,
    TouchableOpacity,
} from 'react-native';


export class ExerciseCard extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <View style={styles.container}>
                <Text style={styles.title}>{this.props.title}</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        backgroundColor: '#fff',
        alignContent: 'stretch',
        marginBottom: 5,
        padding: 10,
    },
    title: {
        fontSize: 18,
    }
});