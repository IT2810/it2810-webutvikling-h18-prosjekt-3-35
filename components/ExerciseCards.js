import React, {Component} from 'react';
import {Card, Title} from "react-native-paper";
import {AsyncStorage, StyleSheet} from "react-native";

export class ExerciseCards extends Component {
    constructor(props) {
        super(props);
        this.state = {
            exercises: [
                {name: 'Markløft'},
                {name: 'Knebøy'},
                {name: 'Benkpress'}],
        }
    }

    // componentDidMount = () => this.retrieveExerciseData('exercises');

    retrieveExerciseData = async (location) => {
        try {
            const value = await AsyncStorage.getItem(location);
            this.setState({
                exercises: JSON.parse(value),
            });
        } catch (error) {
            console.warn(error);
        }
    };

    render() {
        return this.state.exercises.map(ex => (
            <Card style={styles.card} key={ex.name}>
                <Card.Content>
                    <Title>{ex.name}</Title>
                </Card.Content>
            </Card>
        ));
    }
}

const styles = StyleSheet.create({
    card: {
        marginBottom: 2,
    }
});
