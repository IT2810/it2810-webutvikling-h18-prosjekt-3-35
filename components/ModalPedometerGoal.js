import 
    React, {
    Component,
} from 'react';
import {
    StyleSheet,
    Modal,
    View,
    Text,
    TextInput,
    Button,
    Alert,
    Keyboard,
    TouchableWithoutFeedback,
} from 'react-native';

const backButtonText = "Back";
const acceptButtonText = "OK"

export class ModalPedometerGoal extends Component {
    constructor(props) {
        super(props);
        this.state = {
            goal: this.props.goal,
        }
    }

    componentDidUpdate = (prevProps) => {
        if (prevProps.goal !== this.props.goal) {
            this.setState({goal: this.props.goal});
        }
    }
    
    render() {
        if (this.props.visible) {
            return(
                <View>
                    <Modal
                        style={styles.modalContainer}
                        animationType="slide"
                        transparent={false}
                        visible={this.props.visible}
                        onRequestClose={() => {
                            Alert.alert('Modal has been closed.');
                        }}>
                            <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
                                <View style={styles.modalContainer}>
                                    <View style={styles.stateText}>
                                        <Text style={styles.text}>You have walked {this.props.steps} steps</Text>
                                        <Text style={styles.text}>Your daily goal is {this.state.goal} steps</Text>
                                    </View>
                                    <View style={styles.textInputRow}>
                                        <Text>Edit goal steps:</Text>  
                                            <TextInput
                                                style={styles.inputField}
                                                defaultValue={String(this.props.goal)}
                                                keyboardType={'numeric'}
                                                onChangeText={(number) => this.setState({goal: number})}/>                            
                                    </View>
                                    <View style={styles.buttonRow}>
                                        <Button
                                        title={backButtonText}
                                        onPress={() => this.props.hideModal()}
                                        />

                                        <Button 
                                        title={acceptButtonText}
                                        onPress={() => this.props.acceptChange(this.state.goal)}
                                        />
                                    </View>
                                </View>
                            </TouchableWithoutFeedback>
                    </Modal>
                </View>
            );
        }
        return false;
    }
}

const styles = StyleSheet.create({
    modalContainer: {
        flexDirection: 'column',
        flex: 1,
        margin: 20,
    },
    inputField: {
        padding: 5,
        textAlign: 'center',
    },
    textInputRow: {
        flex: 1,
        flexDirection: 'column',
        paddingBottom: 20,
    },
    buttonRow: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        marginBottom: 20,
    },
    stateText: {
        flex: 1,
        marginTop: 20,
        marginBottom: 20,
    },
    text:  {
        textAlign: 'center',
        fontSize: 20,
    }
});