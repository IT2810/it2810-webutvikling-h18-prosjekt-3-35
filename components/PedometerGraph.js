import
React, {
    Component
} from 'react';
import {
    StyleSheet,
    View,
    Text,
} from 'react-native';
import {
    ProgressCircle
} from 'react-native-svg-charts';

export class PedometerProgressGraph extends Component {
    constructor(props) {
        super(props);
        this.state = {
            progress: 0,
        }
    }

    componentDidMount = () => this.calculateProgress(this.props.stepsWalked, this.props.goal);

    calculateProgress = (stepsWalked, goal) => this.setState({
        progress: (stepsWalked / goal)
    });

    goalText = () => {
        let returnText;
        if (this.state.progress === 1 || this.state.progress > 1) {
            returnText = <Text style={styles.resultText}>Du har nådd målet!</Text>
        } else {
            returnText = <View></View>
        }

        return (
            <View>
                <Text
                    style={styles.distance}>
                    {this.props.stepsWalked}
                </Text>
                {returnText}
            </View>
        );
    }

    render() {
        const goalResutText = this.goalText();
        return (
            <View>
                <ProgressCircle
                    style={styles.graph}
                    progress={ this.state.progress }
                    progressColor={'rgb(134, 65, 244)'}
                    startAngle={ -Math.PI * 0.8 }
                    endAngle={ Math.PI * 0.8 }
                />
                {goalResutText}
                <View style={styles.lineSeperator}/>
            </View>
        );
    }

}

const styles = StyleSheet.create({
    graph: {
        height: 130,
    },
    distance: {
        textAlign: 'center',
        fontSize: 24,
    },
    lineSeperator: {
        alignSelf: 'stretch',
        borderWidth: 0.2,
        borderColor: 'black',
        backgroundColor: 'lightgray',
        marginTop: 5,
        marginBottom: 10,
    },
    resultText: {
        textAlign: 'center',
        fontSize: 30,
    }
});