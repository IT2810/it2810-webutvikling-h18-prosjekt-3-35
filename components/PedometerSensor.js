import React, {
    Component
} from 'react';
import { Text, View } from 'react-native';
import { Pedometer, Expo } from 'expo';

export default class PedometerSensor extends Component {
    constructor(props) {
        super(props);
        this.state =  {
            pastStepCount: 0,
            currentStepCount: 0,
            isPedometerAvailable: 'Checking',
        }
    }

    componentDidMount = () => {
        this.subscribe();
        console.log("mounting pedometer");
    }
    
    componentWillUnmount = () => {
        this.unsubscribe();
        console.log("unmounting pedometer");
    }
    
    subscribe = async () => {
        this.subscription = Pedometer.watchStepCount(result => {
            this.setState({currentStepCount: result.steps});
            console.log(this.state.pastStepCount + this.state.currentStepCount);
            this.props.updateSteps(this.state.pastStepCount + this.state.currentStepCount);
        });

        const end = new Date();
        const start = new Date();
        start.setHours(0,0,0,0);
        end.setHours(24,0,0,0);
        console.log(start);
        console.log(end);

        Pedometer.getStepCountAsync(start, end).then(
            result => {
                this.setState({ pastStepCount: result.steps});
                this.props.updateSteps(this.state.pastStepCount);
            },
            error => {
                this.setState({pastStepCount: "Could not get stepCount: " + error});
            }
        );
    };
    
    unsubscribe = () => {
        this.subscription && this.subscription.remove();
        this.subscription = null;
    };

    updateSteps = (steps) => this.props.updateSteps(steps);

    render() {
        return (false);
      }
}