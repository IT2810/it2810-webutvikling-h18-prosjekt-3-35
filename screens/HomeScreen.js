import React from 'react';
import {
    Platform,
    ScrollView,
    Image,
    StyleSheet,
    View,
    TouchableOpacity
} from 'react-native';
import {
    DatePicker
} from '../components/DatePicker';
import {
    PedometerProgressGraph
} from '../components/PedometerGraph';
import {
    ModalPedometerGoal
} from '../components/ModalPedometerGoal';

const logoSource = '../assets/images/pmm.png';

export default class HomeScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            pedometerModalVisible: false,
            stepsWalked: 1000,
            stepGoal: 500,
        }
    }
    static navigationOptions = {
        header: null,
    };


    showHidePedometerModal = () => this.setState({pedometerModalVisible: !this.state.pedometerModalVisible});

    editStepGoal = (goal) => {
        this.setState({
        pedometerModalVisible: !this.state.pedometerModalVisible,
        stepGoal: parseInt(goal, 10),
    })
    console.log(goal);
}

    render() {
        return (
            <View style = {styles.container}>
                <Image 
                    source = {require(logoSource)}
                    style = {styles.logo}/>
                <View style = {styles.lineStyle}/>
                <ScrollView style = {styles.ScrollView} >
                    <TouchableOpacity onPress={() => this.showHidePedometerModal()}>
                        <PedometerProgressGraph 
                            stepsWalked={this.state.stepsWalked} 
                            goal={this.state.stepGoal} />
                    </TouchableOpacity>
                    { /*TODO: put exercise list in here*/ } 
                </ScrollView>
                <ModalPedometerGoal 
                    visible={this.state.pedometerModalVisible}
                    hideModal={this.showHidePedometerModal.bind(this)}
                    acceptChange={this.editStepGoal.bind(this)}
                    goal={this.state.stepGoal}
                    steps={this.state.stepsWalked}/>
            </View>);
        }
    }

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'lightgray',
        justifyContent: 'center',
        alignItems: 'center',
        flex: 1,
        paddingHorizontal: 20,
    },
    logo: {
        marginTop: 18,
        marginBottom: 5,
        height: 100,
        width: 100
    },
    ScrollView: {
        backgroundColor: 'lightgray',
        alignSelf: 'stretch',
    },
    lineStyle: {
        alignSelf: 'stretch',
        borderWidth: 3,
        borderColor: 'black',
        borderBottomColor: 'lightgray',
        backgroundColor: 'lightgray',
        marginBottom: 10,
    }
});