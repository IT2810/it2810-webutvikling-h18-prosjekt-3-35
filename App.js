import React, { Component } from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font, Icon } from 'expo';
import { createStackNavigator } from 'react-navigation';
import {DefaultTheme, Provider as PaperProvider} from 'react-native-paper';

import HomeScreen from './screens/HomeScreen';
import StepGoalScreen from './screens/StepGoalScreen';
import CreateExerciseScreen from './screens/CreateExerciseScreen';
import GraphingScreen from './screens/GraphingScreen';
import CreateSessionScreen from './screens/CreateSessionScreen';

const theme = {
    ...DefaultTheme,
    dark: true,
    roundness: 2,
    colors: {
        ...DefaultTheme.colors,
        primary: '#5c92aa',
        accent: '#5c92aa',
        background: '#ecf8ff',
        surface: '#30375e',
        text: '#041234',
        disabled: '#8dc2dc',
        placeholder: '#5c618b',
        backdrop: '#ffffff',
    },
    fonts: {
        ...DefaultTheme.fonts,

    }
};

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoadingComplete: false,
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />);
    } else {
      return(
          <PaperProvider theme={theme}>
              <RootStack />
          </PaperProvider>
      );
    }
  }
}

const RootStack = createStackNavigator({
    Home: HomeScreen,
    StepGoal: StepGoalScreen,
    CreateExercise: CreateExerciseScreen,
    ExerciseGraph: GraphingScreen,
    CreateSession: CreateSessionScreen,
  },
  {
    initialRouteName: 'Home',
});
