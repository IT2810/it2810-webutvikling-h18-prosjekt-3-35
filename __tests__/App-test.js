import 'react-native';
import React from 'react';
import App from '../App';
import PedometerProgressGraph from '../screens/HomeScreen';
import {stepsWalked} from '../screens/HomeScreen';
import {goal} from '../screens/HomeScreen';
import {goalResutText} from '../components/PedometerGraph';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the loading screen', async () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
    describe('<HomeScreen>', () => {
        it('Test mål, skritt gått, og om pedo er aktiv eller ei', () => {

            const inst = renderer.create(
                <PedometerProgressGraph returnText={stepsWalked}>

                </PedometerProgressGraph>
            );

            let instance = inst.root;

            console.log(instance._fiber.stateNode.state);

             expect(instance._fiber.stateNode.state.stepsWalked).toEqual(1000);
             expect(instance._fiber.stateNode.state.stepGoal).toEqual(10000);
             expect(instance._fiber.stateNode.state.pedometerModalVisible).toEqual(false);

            expect(inst.props).toMatchSnapshot();

        });
    });

  it('renders the root without loading screen', async () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
