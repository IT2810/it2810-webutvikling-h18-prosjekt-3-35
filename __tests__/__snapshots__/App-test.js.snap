// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot <HomeScreen> Test m책l, skritt g책tt, og om pedo er aktiv eller ei 1`] = `undefined`;

exports[`App snapshot <PedometerProgressGraph> Test om m책l n책d 1`] = `undefined`;

exports[`App snapshot renders the loading screen 1`] = `null`;

exports[`App snapshot renders the root without loading screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "marginBottom": 5,
          "marginTop": 18,
          "width": 100,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "lightgray",
          "borderBottomColor": "lightgray",
          "borderColor": "black",
          "borderWidth": 3,
          "marginBottom": 10,
        }
      }
    />
  </View>
</View>
`;
